x-pack/legacy/plugins/lens/public/xy_visualization_plugin/xy_visualization.tsx

```diff
@@ -5,17 +5,75 @@
  */
 
 import React from 'react';
+import _ from 'lodash';
 import { render } from 'react-dom';
 import { Position } from '@elastic/charts';
 import { I18nProvider } from '@kbn/i18n/react';
+import { i18n } from '@kbn/i18n';
+import { EuiIcon } from '@elastic/eui';
 import { getSuggestions } from './xy_suggestions';
 import { XYConfigPanel } from './xy_config_panel';
 import { Visualization } from '../types';
-import { State, PersistableState } from './types';
+import { State, PersistableState, SeriesType, visualizationTypes } from './types';
 import { toExpression } from './to_expression';
 import { generateId } from '../id_generator';
 
+const defaultIcon = 'visBarVertical';
+const defaultSeriesType = 'bar';
+
+function getDescription(state?: State) {
+  if (!state) {
+    return {
+      icon: defaultIcon,
+      label: i18n.translate('xpack.lens.xyVisualization.xyLabel', {
+        defaultMessage: 'XY Chart',
+      }),
+    };
+  }
+
+  if (!state.layers.length) {
+    return visualizationTypes.find(v => v.id === state.preferredSeriesType)!;
+  }
+
+  const visualizationType = visualizationTypes.find(t => t.id === state.layers[0].seriesType)!;
+  const seriesTypes = _.unique(state.layers.map(l => l.seriesType));
+
+  return {
+    icon: visualizationType.icon,
+    label:
+      seriesTypes.length === 1
+        ? visualizationType.label
+        : i18n.translate('xpack.lens.xyVisualization.mixedLabel', {
+            defaultMessage: 'Mixed XY Chart',
+          }),
```

Comment:
Oh, my earlier comments did not consider this case. I think this requires a function, but not necessarily a render function.

Reply:
Changed it to a function `(state) => {label: string; icon: string }`